lib/tests/misc.nix

```
@@ -894,6 +895,43 @@ runTests {
     expected = { a = [ 2 3 ]; b = [7]; c = [8];};
   };
 
+  testFlattenAttrs = {
+    expr = flattenAttrs {
+        a.b.c.d = "e";
+        one.two.three.four.five = 6;
+        foo.bar = null;
+        yes.no.maybe = true;
+      };
+    expected = {
+      d = "e";
+      five = 6;
+      bar = null;
+      maybe = true;
+    };
+  };
+
+  testFlattenAttrsEmptyAttrs = {
+    expr = flattenAttrs { };
+    expected = { };
+  };
+
```

Maybe some test for conflicting bottom keys ? What would be the desired behavior?

```suggestion
flattenAttrs {
        a.foo = "A";
        b.foo = "B";
      };
```

---

This is a good mention. This was mentioned in *Points of Consideration*. Seeing as the function itself needs some reimplementation, and even potentially some branching into some other function variations, I'll rework the test cases once that is done first. I will revisit this.